### ISSUE TITLE
[mnewcomb on 2011-02-11 16:07:37] : pychecker complaints on dash code with comments

### ISSUE THREAD

**Issue by dglo** [dglo/pdaq#1590]
# might be bad as one return is a float and the other is an int
LogSummary.py:780: Function return types are inconsistent

# not sure, but "in-ice" or "icetop" are turned as a tuple.
# the amanda one is a string ( no tuple )
DeployPDAQ.py:43: Function return types are inconsistent
DeployPDAQ.py:44: Function return types are inconsistent

# valid problem
CnCRunSetTest.py:35: No class attribute (type) found
CnCRunSetTest.py:37: No class attribute (type) found

# just a typo instead of runNum should be self.__runNum
BaseRun.py:173: No global (runNum) found

# another missing variable
DAQConfig.py:562: No global (fileName) found

# huh? would have expected rd.recv()
DAQFakeRun.py:113: No global (sock) found

# the next seven errors are due to an addDOM method in the SubRun class
ExpControlSkel.py:133: No global (bright) found
ExpControlSkel.py:133: No global (delay) found
ExpControlSkel.py:133: No global (pos) found
ExpControlSkel.py:133: No global (string) found
ExpControlSkel.py:133: No global (window) found
ExpControlSkel.py:134: No global (mask) found
ExpControlSkel.py:134: No global (rate) found

# I think this is just a missing re. or import issue
ExpControlSkel.py:159: No global (search) found

# errors in the main() function for ExpControlSkel
ExpControlSkel.py:248: No global (flashPause) found
ExpControlSkel.py:248: No global (flashTimes) found

# missing import
ExpControlSkel.py:33: No global (datetime) found

# bad
liverun.py:624: No global (runNum) found

# not sure this package is used anymore
IntegrationTest.py:1135: No global (dr) found
IntegrationTest.py:812: No global (MostlyLive) found
# missing import
CnCServerTest.py:78: No global (MockLogger) found

# just a typo, should be self.fail
RunClusterTest.py:131: No global (fail) found

# not sure this is used as it's not on dave's list of interesting stuff
IntegrationTest.py:1114: Object (msgList) has no attribute (append)

# both true, self instead.
CnCRunSetTest.py:632: ['cls', 'klass'] is not first classmethod argument
DAQConfig.py:366: ['cls', 'klass'] is not first classmethod argument

# looks like this wants to be a classmethod but it isn't declared so
pdash.py:25: self is not first method argument

# problem in main
CreateClusterConfig.py:102: No class attribute (fileExists) found

# '__' in a variable name starts python's name mangler.
# Python mangles these names with the class name: if class Foo has an
#      attribute named __a, it cannot be accessed by Foo.__a.  (An insistent
#      user could still gain access by calling Foo._Foo__a.)
DAQFakeRun.py:242: No class attribute (_FakeConnector__name) found

# missing argument
CreateClusterConfig.py:114: Invalid arguments to (load), got 2, expected between
 3 and 4

# valid problem
FakeRunset.py:51: Invalid arguments to (createHubs), got 3, expected between 4 a
nd 5

# missing an argument, good complaint
DAQMocks.py:1171: Format string argument count (4) doesn't match arguments (5)

# All of the below are from putting a , at the end of a parse.add_option
# statement.  Good complaint?
DAQFakeRun.py:1444: Statement appears to have no effect
DAQFakeRun.py:1447: Statement appears to have no effect
DAQFakeRun.py:1459: Statement appears to have no effect
DAQFakeRun.py:1465: Statement appears to have no effect
DAQFakeRun.py:1468: Statement appears to have no effect
DAQFakeRun.py:1471: Statement appears to have no effect
DAQFakeRun.py:1474: Statement appears to have no effect
FakeRunset.py:11: Statement appears to have no effect
FakeRunset.py:14: Statement appears to have no effect
FakeRunset.py:17: Statement appears to have no effect
FakeRunset.py:20: Statement appears to have no effect
FakeRunset.py:23: Statement appears to have no effect

# seem valid
DAQLiveTest.py:145: Module (traceback) re-imported
ExpControlSkel.py:11: Module (optparse) re-imported
ExpControlSkel.py:13: Module (sys) re-imported
IntegrationTest.py:1057: Module (datetime) re-imported
CnCServer.py:1273: Module (signal) re-imported
SpadeQueue.py:101: No module attribute (utcnow) found
SpadeQueue.py:87: Using import and from ... import for (datetime)
UploadHubs.py:15: DeprecationWarning: The popen2 module is deprecated.  Use the
subprocess module.

**Comment by dglo** [comment]
[mnewcomb on 2011-02-11 22:13:14]
Returned 0.0 when there is nothing in the log to keep return types consistent
LogSummary.py:780: Function return types are inconsistent

This method used to return either a tuple or a string due to an extra comma after some return values
Changed to remove the extra commas
DeployPDAQ.py:43: Function return types are inconsistent
DeployPDAQ.py:44: Function return types are inconsistent

Type did not exist, but there was a __descCh variable and a __name which where both added to the __repr__ for this class
CnCRunSetTest.py:35: No class attribute (type) found
CnCRunSetTest.py:37: No class attribute (type) found

Fixed..  Changed runNum to be initialized to 0 instead of None
Error messages used a %d to format the run number, if it was None it would cause another exception
BaseRun.py:173: No global (runNum) found

Changed to use baseName instead of fileName ( baseName is a parameter and is used as a file name later )
DAQConfig.py:562: No global (fileName) found

Changed error messages to avoid these problems
DAQConfig.py:927: Format string argument count (3) doesn't match arguments (2)
DAQConfig.py:931: Format string argument count (3) doesn't match arguments (2)

A missing exception.  There was one called DAQConfigException so I used that instead of creating a new one.
DAQConfig.py:1137: No global (DAQConfigDirNotFound) found

rd is a list of sockets, and recv has to be called on each.  Loop through all readable sockets and call recv on them
DAQFakeRun.py:113: No global (sock) found

Not sure where these parameters where supposed to come from so commented out this call and added a NotImplementedError exception
ExpControlSkel.py:133: No global (bright) found
ExpControlSkel.py:133: No global (delay) found
ExpControlSkel.py:133: No global (pos) found
ExpControlSkel.py:133: No global (string) found
ExpControlSkel.py:133: No global (window) found
ExpControlSkel.py:134: No global (mask) found
ExpControlSkel.py:134: No global (rate) found

There where a number of these in the __init__ for a class.  Obviously not used for a long time, so a comment was added to the class and the missing re.'s where added
ExpControlSkel.py:159: No global (search) found

It appears that the structure for starting flasher runs may have changed.  These are arguments to get a flasher run started and they don't exist
For now if you try to start a flasher run with ExpControlSkel.py you will get an error message and it will exit cleanly.
ExpControlSkel.py:248: No global (flashPause) found
ExpControlSkel.py:248: No global (flashTimes) found

added the missing import
ExpControlSkel.py:33: No global (datetime) found

added a call to getRunNum to get the run number, surrounded by a try/except.  If there is an exception the error message just does not include a run number
liverun.py:624: No global (runNum) found

The section this variable was in is not used anymore and was removed as per Dave G.
IntegrationTest.py:1135: No global (dr) found

Removed
IntegrationTest.py:3: Imported module (StringIO) not used
IntegrationTest.py:3: Imported module (re) not used

Do not get rid of the code here as per Dave G.  Instead I created a class that raises a NotImpementedError in the __init__ method
IntegrationTest.py:812: No global (MostlyLive) found

added missing import
CnCServerTest.py:78: No global (MockLogger) found

This was supposed to be self.fail as per the surrounding calls.  Changed
RunClusterTest.py:131: No global (fail) found

I changed msgList to a list instead of a tuple so append would work.
IntegrationTest.py:1114: Object (msgList) has no attribute (append)

Changed from self to cls
CnCRunSetTest.py:632: ['cls', 'klass'] is not first classmethod argument

Changed from self to cls
DAQConfig.py:366: ['cls', 'klass'] is not first classmethod argument

looks like this wants to be a classmethod but it isn't declared so I made it a class method
pdash.py:25: self is not first method argument

Looking for a method called fileExists..  Not there, but there is one called configExists..  Changed to one.
configExists has a keyword argument for the configuration directory.  Load below wants a directory but has no keyword arg
To stay consistent build a directory and use it in both places
CreateClusterConfig.py:102: No class attribute (fileExists) found
CreateClusterConfig.py:114: Invalid arguments to (load), got 2, expected between 3 and 4

Instead of the connector name, used a string representation class for the connector to get a description
DAQFakeRun.py:242: No class attribute (_FakeConnector__name) found


Extra argument, numEvents is none on this line.  Fixed
DAQFakeRun.py:1239: Format string argument count (2) doesn't match arguments (3)

The missing argument was a "sendTrackHits" which looks to be related to the track engine.  In DAQFakeRun.ComponentData.createAll a keyword argument is used for this and it defaults to false so I'm considering it safe to send a false here. 
FakeRunset.py:51: Invalid arguments to (createHubs), got 3, expected between 4 and 5

Typo in the format string, should have been # instead of % ( made the error message look like surrounding ones )
DAQMocks.py:1171: Format string argument count (4) doesn't match arguments (5)

Removed extra comma's and the complaints are gone
DAQFakeRun.py:1444: Statement appears to have no effect
DAQFakeRun.py:1447: Statement appears to have no effect
DAQFakeRun.py:1459: Statement appears to have no effect
DAQFakeRun.py:1465: Statement appears to have no effect
DAQFakeRun.py:1468: Statement appears to have no effect
DAQFakeRun.py:1471: Statement appears to have no effect
DAQFakeRun.py:1474: Statement appears to have no effect
FakeRunset.py:11: Statement appears to have no effect
FakeRunset.py:14: Statement appears to have no effect
FakeRunset.py:17: Statement appears to have no effect
FakeRunset.py:20: Statement appears to have no effect
FakeRunset.py:23: Statement appears to have no effect


Removed extra imports
DAQLiveTest.py:145: Module (traceback) re-imported
ExpControlSkel.py:11: Module (optparse) re-imported
ExpControlSkel.py:13: Module (sys) re-imported
IntegrationTest.py:1057: Module (datetime) re-imported
CnCServer.py:1273: Module (signal) re-imported
SpadeQueue.py:101: No module attribute (utcnow) found
SpadeQueue.py:87: Using import and from ... import for (datetime)

Left for now
UploadHubs.py:15: DeprecationWarning: The popen2 module is deprecated.  Use the
subprocess module.

Moved definition of waitList farther up in the method
RunSet.py:782: Variable (waitList) used before being set

Removed as per Dave G.
RunCluster.py:335: No class attribute (_RunCluster__hostName) found
RunCluster.py:336: No class attribute (_RunCluster__locName) found

This was a typo in a method that was never called ( the call was commented out ) so it was removed as per Dave G.
RunCluster.py:261: No global (obj) found

Error was used where there should be an exception.  Changed to ValueError
CnCServer.py:467: No global (Error) found
